android/anc/src/main/java/org/smartregister/fhircore/anc/data/patient/PatientRepository.kt

```diff
@@ -220,6 +220,7 @@ constructor(
       when (searchFilterString) {
         "body-weight" -> vitalSignsConfig.weightFilter!!
         "body-height" -> vitalSignsConfig.heightFilter!!
+        "bmi" -> vitalSignsConfig.bmiFilter!!
         "bp-s" -> vitalSignsConfig.BPSFilter!!
         "bp-d" -> vitalSignsConfig.BPDSFilter!!
         "pulse-rate" -> vitalSignsConfig.pulseRateFilter!!
```

Comment:
I think you can use the `run` function to remove the redundant `vitalSignConfig` variable.

Reply:
updated

Reply:
The correct scope function in this use case is `with`. `run` is used when you want to call member functions on the object instance accessible using `this`. `with` scope function also allows you to access the member properties of the class/object.